localstack/aws/connect.py

```diff
@@ -0,0 +1,295 @@
+"""
+LocalStack client stack.
+
+This module provides the interface to perform cross-service communication between
+LocalStack providers.
+"""
+import json
+from functools import cache
+from typing import Mapping, Optional, TypedDict, Union
+
+from boto3.session import Session
+from botocore.awsrequest import AWSPreparedRequest
+from botocore.client import BaseClient
+from botocore.config import Config
+from werkzeug.datastructures import Headers
+
+from localstack import config
+from localstack.constants import (
+    INTERNAL_AWS_ACCESS_KEY_ID,
+    INTERNAL_AWS_SECRET_ACCESS_KEY,
+    MAX_POOL_CONNECTIONS,
+)
+from localstack.utils.aws.arns import extract_region_from_arn
+from localstack.utils.aws.aws_stack import get_local_service_url
+from localstack.utils.aws.request_context import get_region_from_request_context
+
+#
+# Data transfer object
+#
+
+LOCALSTACK_DATA_HEADER = "x-localstack-data"
+"""Request header which contains the data transfer object."""
+
+
+class LocalStackData(TypedDict):
```

Comment:
nit: i don't think this name is very descriptive. i think something like `ClientRequestParameters` or `LocalstackRequestParameters` would help a reader understand more what this is.

Reply:
Renamed to `LocalstackRequestParameters`